### PR TITLE
feat(krieger): Augment owner resolution strategy with krieger

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Added new module `swabbie-krieger` (can be in -nflx)
- Update owner resolution to return a comma delimited owners
- comma delimited owners ensures we dont miss an owner in strategies
- prefer notifying a group to only notifying a subset if there are > 1 owners

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Implements https://github.com/spinnaker/swabbie/issues/93.
Currently image owners are derived from Tags which is not sufficient at Netflix because the `creator|owner` tag values points to the person that may baked the image. This ensures the actual owners of applications using the ami are notified.
@emjburns could you ensure notifications can be sent to a comma delimited email field in echo: something like `lucious-mayweather@netflix.com, jiles-tattoo@netflix.com` but with real email addresses :)

File: swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/ResourceOwnerResolver.kt [hunk 1]
```diff
@@ -71,5 +71,7 @@ interface OwnerResolver<in T : Resource> {
 }
 
 interface ResourceOwnerResolutionStrategy<in T : Resource> {
+  val log: Logger
+    get() = LoggerFactory.getLogger(javaClass)
```
[the remark this answers is not in the document]
Comment (jeyrschabu) [reply swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/ResourceOwnerResolver.kt]: From what I understand, Kotlin adds getters to all properties regardless of if you define them or not.  In my understanding, the impl classes of this interface will get a log object when they are loaded by the jvm, log is static so I don’t expect to get a different instance each time log is invoked. @robzienert is my understanding incorrect? If so please update the PR or @emburns can do so. My laptop is officially put away:)

**Comment by jeyrschabu** [comment]
@robzienert RE:nflx vs OSS, I tend to agree with your approach and I don’t have a strong opinion on either option. I will leave it up to @emburns judgment. If we decide to nflx it, I would recommend splitting this resolution keeping the original in OSS with the new change to include all owners. The internal one can be Krieger specific.

**Comment by jeyrschabu** [comment]
@emburns could you follow up on this PR and any other. Feel free to do as you please. Merge them, close them, update them... My lappy has been tuck away, soooo ....

**Comment by emjburns** [comment]
@jeyrschabu @robzienert I think I'd like to pull this into `-nflx`. I'll open a seperate PR for the owner -> owners. I'll apply rob's feedback internally. Will close this once I've created the other PRs.

**Comment by emjburns** [comment]
Closing as functionality has been implemented in separate PRs.

**Pull request closed without merging**